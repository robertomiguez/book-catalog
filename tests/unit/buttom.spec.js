import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import Button from '@/components/Button.vue'

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Button, {
    propsData: { 
      to: `somewhere`, 
      text: `click` 
    },
    mocks: {},
    stubs: {
      RouterLink: RouterLinkStub
    },
    methods: {}
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe(`Buttom`, () => {
  test(`testing component props`, () => {
    expect(wrapper.find(`.button`).text())
      .toBe(`click`)
    expect(wrapper.find(RouterLinkStub).props().to)
      .toEqual({ "name": `somewhere` })
  })
    
  test(`regression test`, () => {
    expect(wrapper.element)
      .toMatchSnapshot()
  })  

})