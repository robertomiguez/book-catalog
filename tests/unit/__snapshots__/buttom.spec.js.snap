// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttom regression test 1`] = `
<div>
  <a
    class="button"
  >
    
		click
	
  </a>
</div>
`;
